feature_engine/base_transformers.py

```diff
@@ -144,3 +144,26 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
     # for the check_estimator tests
     def _more_tags(self):
         return _return_tags()
+
+    def get_feature_names(
+        self,
+        input_features: List[str] = None,
+    ) -> List:
+        """
+        Return feature names for output features.
+
+        Parameters
+        ----------
+        input_features : list of str of shape (n_features,)
+            String names for input features if available. By default,
+            self.variables_ is used
+
+        -------
+        feature_names : list of str of shape (n_output_features,) of feature names
+        """
+        check_is_fitted(self)
+
+        if input_features is None:
+            input_features = self.variables_
+
+        return [feature for feature in input_features if feature in self.variables_]
```

Comment:
I don't think this would capture the names of all features in the final transformed dataframe. Is this the desired output of this method? maybe I did not understand the aim of the function?

Reply:
I thought that for the generic case this would capture output features. Am I missing something?